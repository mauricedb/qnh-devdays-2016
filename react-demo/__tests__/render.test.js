var React = require('react');
var renderer = require('react-test-renderer');

var MoviesList = require('../app/moviesList.jsx');
var MovieEditor = require('../app/movieEditor.jsx');

var InputText = require('../app/inputText.jsx');
var TextArea = require( '../app/textArea.jsx');

describe('All components should render correctly', () => {
    it('empty MoviesList', () => {
        const tree = renderer.create(
            <MoviesList movies={[]} 
                        toEditMode={() => {}} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('non empty MoviesList', () => {
        const tree = renderer.create(
            <MoviesList  movies={[{
                id: '1',
                title: 'Kill Bill',
                abridgedDirectors: ['Quentin Tarantino']
            }]}
            toEditMode={() => {}}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('empty MovieEditor', () => {
        const tree = renderer.create(
            <MovieEditor movie={{
                            id: '',
                            title: '',
                            abridgedDirectors: [],
                            criticsConsensus: '',
                            synopsis: '',
                            year: 0
                         }} 
                         onSave={() => {}}
                         toListMode={() => {}}
                         onChange={() => {}} 
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('non empty MovieEditor', () => {
        const tree = renderer.create(
            <MovieEditor movie={{
                            id: '1',
                            title: 'Kill Bill',
                            abridgedDirectors: ['Quentin Tarantino'],
                            criticsConsensus: 'Good',
                            synopsis: 'A revenge movie',
                            year: 2003
                         }} 
                         onSave={() => {}}
                         toListMode={() => {}}
                         onChange={() => {}} 
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('InputText', () => {
        const tree = renderer.create(
            <InputText value={'Kill Bill'}
                        prop='title'
                        onChange={() => {}}>
                Title
            </InputText>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('TextArea', () => {
        const tree = renderer.create(
            <TextArea value={'movie.criticsConsensus'}
                    prop='criticsConsensus'
                    onChange={() => {}}>
                Critics Consensus
            </TextArea>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

