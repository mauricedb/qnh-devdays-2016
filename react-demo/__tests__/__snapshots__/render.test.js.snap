exports[`All components should render correctly InputText 1`] = `
<div
  className="form-group">
  <label>
    Title
  </label>
  <input
    className="form-control"
    onChange={[Function onChange]}
    type="text"
    value="Kill Bill" />
</div>
`;

exports[`All components should render correctly TextArea 1`] = `
<div
  className="form-group">
  <label>
    Critics Consensus
  </label>
  <textarea
    className="form-control"
    onChange={[Function onChange]}
    rows="5"
    value="movie.criticsConsensus" />
</div>
`;

exports[`All components should render correctly empty MovieEditor 1`] = `
<form>
  <div
    className="form-group">
    <label>
      Title
    </label>
    <input
      className="form-control"
      onChange={[Function onChange]}
      type="text"
      value="" />
  </div>
  <div
    className="form-group">
    <label>
      Directors
    </label>
    <input
      className="form-control"
      onChange={[Function onChange]}
      type="text"
      value="" />
  </div>
  <div
    className="form-group">
    <label>
      Critics Consensus
    </label>
    <textarea
      className="form-control"
      onChange={[Function onChange]}
      rows="5"
      value="" />
  </div>
  <div
    className="form-group">
    <label>
      Synopsis
    </label>
    <textarea
      className="form-control"
      onChange={[Function onChange]}
      rows="5"
      value="" />
  </div>
  <div
    className="form-group">
    <label>
      Year
    </label>
    <input
      className="form-control"
      onChange={[Function bound onChange]}
      type="text"
      value="" />
  </div>
  <div
    className="form-group">
    <button
      className="btn btn-primary"
      onClick={[Function bound save]}>
      Save
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function bound toListMode]}>
      Cancel
    </button>
  </div>
</form>
`;

exports[`All components should render correctly empty MoviesList 1`] = `
<table
  className="table table-bordered table-striped">
  <thead>
    <tr>
      <th>
        Title
      </th>
      <th>
        Directors
      </th>
      <th />
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`All components should render correctly non empty MovieEditor 1`] = `
<form>
  <div
    className="form-group">
    <label>
      Title
    </label>
    <input
      className="form-control"
      onChange={[Function onChange]}
      type="text"
      value="Kill Bill" />
  </div>
  <div
    className="form-group">
    <label>
      Directors
    </label>
    <input
      className="form-control"
      onChange={[Function onChange]}
      type="text"
      value="Quentin Tarantino" />
  </div>
  <div
    className="form-group">
    <label>
      Critics Consensus
    </label>
    <textarea
      className="form-control"
      onChange={[Function onChange]}
      rows="5"
      value="Good" />
  </div>
  <div
    className="form-group">
    <label>
      Synopsis
    </label>
    <textarea
      className="form-control"
      onChange={[Function onChange]}
      rows="5"
      value="A revenge movie" />
  </div>
  <div
    className="form-group">
    <label>
      Year
    </label>
    <input
      className="form-control"
      onChange={[Function bound onChange]}
      type="text"
      value={2003} />
  </div>
  <div
    className="form-group">
    <button
      className="btn btn-primary"
      onClick={[Function bound save]}>
      Save
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function bound toListMode]}>
      Cancel
    </button>
  </div>
</form>
`;

exports[`All components should render correctly non empty MoviesList 1`] = `
<table
  className="table table-bordered table-striped">
  <thead>
    <tr>
      <th>
        Title
      </th>
      <th>
        Directors
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Kill Bill
      </td>
      <td>
        Quentin Tarantino
      </td>
      <td
        style={
          Object {
            "width": 1
          }
        }>
        <button
          className="btn btn-default btn-xs edit-button"
          onClick={[Function onClick]}>
          Edit
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
